near/omni-token/src/lib.rs

```diff
@@ -179,39 +175,22 @@ impl MetadataManagment for OmniToken {
     }
 }
 
-#[near]
-impl UpgradeAndMigrate for OmniToken {
-    fn upgrade_and_migrate(&self) {
-        self.assert_controller();
-
-        // Receive the code directly from the input to avoid the
-        // GAS overhead of deserializing parameters
-        let code = env::input().unwrap_or_else(|| env::panic_str("ERR_NO_INPUT"));
-        // Deploy the contract code.
-        let promise_id = env::promise_batch_create(&env::current_account_id());
-        env::promise_batch_action_deploy_contract(promise_id, &code);
-        // Call promise to migrate the state.
-        // Batched together to fail upgrade if migration fails.
-        env::promise_batch_action_function_call(
-            promise_id,
-            "migrate",
-            &json!({ "from_version": CURRENT_STATE_VERSION })
-                .to_string()
-                .into_bytes(),
-            NO_DEPOSIT,
-            env::prepaid_gas()
-                .saturating_sub(env::used_gas())
-                .saturating_sub(OUTER_UPGRADE_GAS),
-        );
-        env::promise_return(promise_id);
-    }
-}
-
 #[near]
 impl FungibleTokenCore for OmniToken {
     #[payable]
     fn ft_transfer(&mut self, receiver_id: AccountId, amount: U128, memo: Option<String>) {
-        self.token.ft_transfer(receiver_id, amount, memo);
+        // Legacy bridging flow used by Near Intents
+        if receiver_id == env::current_account_id()
+            && memo
+                .as_deref()
+                .map_or(false, |memo| memo.starts_with(WITHDRAW_MEMO_PREFIX))
+        {
+            if let Some(withdraw_relayer) = self.read_withdraw_relayer_address() {
```

Comment:
It will be good to have a setter for this in case if we want to update the relayer account.